test/unit/os/env_spec.lua

```diff
@@ -127,4 +129,60 @@ describe('env function', function()
       eq(hostname, (ffi.string(hostname_buf)))
     end)
   end)
+
+  function vim_setenv(name, value)
+    return env.vim_setenv((to_cstr(name)), (to_cstr(value)))
+  end
+
+  describe('vim_setenv', function()
+    it('converts $VIMRUNTIME to $NVIMRUNTIME', function()
+      local old_runtime = os.getenv('NVIMRUNTIME')
+      local runtime = 'set as VIMRUNTIME'
+      vim_setenv('VIMRUNTIME', runtime)
+      eq(nil, os.getenv('VIMRUNTIME'))
+      eq(runtime, os.getenv('NVIMRUNTIME'))
+      -- Restore
+      vim_setenv('NVIMRUNTIME', old_runtime)
```

Comment:
I think this will not be executed if a previous assertion fails. Maybe use a `setup`/`cleanup` function for the whole `describe` block to save/restore environment variables (if that's supported in busted)?